control-plane/subcommand/inject-connect/command.go

```diff
@@ -410,13 +349,13 @@ func (c *Command) Run(args []string) int {
 		c.UI.Error(fmt.Sprintf("unable to create config for consul-server-connection-manager: %s", err))
 		return 1
 	}
-	watcher, err := discovery.NewWatcher(ctx, serverConnMgrCfg, hcLog)
+
+	resourceClient, watcher, err := consul.NewResourceServiceClient(ctx, serverConnMgrCfg, hcLog)
 	if err != nil {
 		c.UI.Error(fmt.Sprintf("unable to create Consul server watcher: %s", err))
 		return 1
 	}
 
-	go watcher.Run()
 	defer watcher.Stop()
```

Comment:
Not certain I'm seeing the whole picture, but: it seems maybe ideal to keep the `Run()` and `Stop()` calls here adjacent to simplify reasoning about lifecycle, and not to duplicate them inside `NewResourceServiceClient`. Could we move the watcher init code back here, or to a helper that could be called independent of client construction, and have `NewResourceServiceClient` be only responsible for constructing the `resourceClient` from a passed `watcher`?

Also curious: is it necessary to create the `resourceClient` outside the v2 controller init? It seems like maybe we could push that concern down rather than handling part of it here, unless there's some future use we anticipate in the "common" code here.

Reply:
Yes, I agree. I didn't anticipate sharing the watcher around in the previous PR. It makes to have it create it outside of the resourceClient and push creating the that to the V2 controller logic.